src/utils.ts

```diff
@@ -25,19 +25,37 @@ export function validateAddress(address: String){
 	return true;
 }
 
-export function nextLinkAddress(address: Byte[]){
-	if(address[1] - 8 > 0){
-		address[1] = address[1] - 8 as Byte;
-	}else{
-		address[0] = address[0] - 1 as Byte;
-		address[1] = 255;
-	}
+export function nextLinkAddress(address: Byte[]) {
+	var next = ((address[0] << 8 | address[1]) - 8);
+	var result = [next >> 8, next & 255] as Byte[];
 
-	if(address[0] < 0){
+	if (result[0] < 0) {
 		throw Error("Out of address space");
 	}
 
-	return address;
+	return result;
+}
+
+export function calculateHighWaterAddress(links: DeviceLinkRecord[]){
```

Comment:
Takes an array of links and determines which link address should be the high water mark (meaning there aren no active links after it)